src/systems/user_commands/UserCommands.cc

```diff
@@ -222,6 +232,26 @@ class ignition::gazebo::systems::UserCommandsPrivate
   public: std::mutex pendingMutex;
 };
 
+/// \brief Pose3d equality comparison function.
+/// \param[in] _a A pose to compare
+/// \param[in] _b Another pose to compare
+bool pose3Eql(const math::Pose3d &_a, const math::Pose3d &_b)
+{
+  return _a.Pos().Equal(_b.Pos(), 1e-6) &&
+    math::equal(_a.Rot().X(), _b.Rot().X(), 1e-6) &&
+    math::equal(_a.Rot().Y(), _b.Rot().Y(), 1e-6) &&
+    math::equal(_a.Rot().Z(), _b.Rot().Z(), 1e-6) &&
+    math::equal(_a.Rot().W(), _b.Rot().W(), 1e-6);
+}
+
+/// \brief Update pose for a specific pose message
+/// \param[in] _req
+/// \param[in] _iface Pointer to user commands interface.
+/// \return True if successful.
+bool UpdatePose(
```

Comment:
We usually use `camelCase` with non-member functions to differentiate them from members (like `pose3Eql` above), but I just realized this isn't documented anywhere.

```suggestion
bool updatePose(
```